File: Data Structures & Algorithms/is-anagram/submission-1.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
            
        mp = {}
        for x in s:
            mp[x] = mp.get(x, 0) + 1
            
        for x in t:
            if x not in mp:
                return False
            mp[x] -= 1
            if mp[x] < 0:
                return False
                
        return True

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        // o(n)
        if(nums.size() == 0) return 0;
        unordered_set<int>st(begin(nums) , end(nums));
        int ans = 1;
        for(auto x : st){
            if(st.find(x-1) == st.end()){
                int streak = 1;
                int num = x;
                while(st.find(num + 1) != st.end())
                {
                    num+=1;
                    streak+=1;
                }
                ans = max(ans , streak);
            }
        }
        return ans;
    }
};